lib/SILOptimizer/Utils/CanonicalizeInstruction.cpp

```diff
@@ -0,0 +1,176 @@
+//===--- CanonicalizeInstruction.cpp - canonical SIL peepholes ------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+// SSA-peephole transformations that yield a more canonical SIL representation.
+//
+// A superset of simplifyInstruction.
+//
+//===----------------------------------------------------------------------===//
+
+#define DEBUG_TYPE "sil-canonicalize"
+
+#include "swift/SILOptimizer/Utils/CanonicalizeInstruction.h"
+#include "swift/SIL/DebugUtils.h"
+#include "swift/SIL/Projection.h"
+#include "swift/SIL/SILBuilder.h"
+#include "swift/SIL/SILFunction.h"
+#include "swift/SILOptimizer/Analysis/SimplifyInstruction.h"
+#include "llvm/ADT/Statistic.h"
+#include "llvm/Support/Debug.h"
+
+using namespace swift;
+
+STATISTIC(NumSimplified, "Number of instructions simplified");
+
+// Tracing within the implementation can also be activiated by the pass.
+#undef DEBUG_TYPE
```

Comment:
Can you document here the header that we want to have the original DEBUG_TYPE?

Reply:
I don't know what you want but I did something more explicit.

Reply:
The new implementation was reviewed in person.